### ISSUE TITLE
Drop bower.js and jQuery

### ISSUE THREAD

**Issue by zooks** [FrDH/jQuery.mmenu#815]
Hi, I think you should drop off obsolete technologies like Bower and jQuery.
Yarn and vanilla.js would be a good replacement.

**Comment by FrDH** [comment]
I myself never used Bower, just added the bower.json file after someone made a pull request. My guess is than when I remove it, I'll get a new pull request to add it :)

I'm not familiair with Yarn, but I think you should already be able to use it: https://yarnpkg.com/en/package/jquery.mmenu

I don't think I'll stick to jQuery for ever, but for now using jQuery has more pros than cons.

**Comment by zooks** [comment]
Thanks for so fast reply.

Bower is outdated: https://bower.io/blog/2017/how-to-migrate-away-from-bower/
So no need to store this file anymore.

Yarn is a replacement for classic npm. Not sure, I think you should to add yarn.lock file to your project. https://github.com/facebook/react/blob/master/yarn.lock
https://yarnpkg.com/lang/en/docs/yarn-lock/

IMO, jQuery means redundant dependency. It will be great if you could get rid of it in the future.

**Comment by FrDH** [comment]
Did a quick sweep through the JS, here's a list of all jQuery methods used by the plugin.
I doubt dropping jQuery as a dependency is a good idea...
````js
$.extend
$.merge
$.grep
$.trim

$(".sel")
$("<a />")

$().length
$().get

$().addClass
$().removeClass

$().attr
$().removeAttr
$().data
$().html
$().text
$().val
$().empty

$().height
$().width
$().innerHeight
$().outerHeight
$().css
$().scrollTop
$().scrollLeft
$().position
$().offset

$().clone

$().replaceWith
$().wrap
$().wrapInner
$().wrapAll
$().insertBefore
$().insertAfter
$().append
$().prepend
$().appendTo
$().prependTo
$().remove

$().find
$().prev
$().prevAll
$().next
$().nextAll
$().children
$().parent
$().parents
$().closest
$().add
$().not
$().filter
$().first
$().last
$().end

$().hasClass
$().is
$().each
$().slice

$().animate

$().on
$().off
$().load
$().trigger
$().focus
$().blur
$().submit
````

**Comment by fatcrobat** [comment]
@FrDH I know this task is already closed, but whats about using https://umbrellajs.com instead of jQuery? You dont have to invent the wheel again completely new, as the syntax is based on jQuery but:

umbrellajs 3.1.0 ~ 2.5kb when gzipped
jQuery 3.3.1- ~30kb when gzipped

Maybe you will an eye on umbrellajs.

**Comment by FrDH** [comment]
great idea

**Comment by FrDH** [comment]
Just as an update: 
I would like to move away from jQuery completely, so I started replacing jQuery methods with vanilla JS methods (for example, using properties or `getAttribute` and `setAttribute` instead of `attr`).
For the DOM traversing and DOM manipulation methods, you'll be able replace jQuery with umbrella.
If you want to have a peek at the new version, check the [develop branche](https://github.com/FrDH/jQuery.mmenu/tree/develop)

**Comment by lkraav** [comment]
I also discovered https://github.com/kenwheeler/cash but maybe vanilla all the way, regardless?

**Comment by FrDH** [comment]
If you want, you could now download the develop branche and replace jQuery with umbrella or cash in the demo files. I think both should work.

**Comment by FrDH** [comment]
I had a closer look at umbrella, and don't think it'll be compattible as a optional replacement for JQuery. As mentioned in their migration guide:
> zepto.js and UmbreallyJS does not support jQuery extensions! if your scripts make heavy use of extensions you should stop reading here.
Also, some basic jQuery logic is implemented differently:
````js
$(".asdf").first(); // collection containing the first HTML node
u(".asdf").first(); // the first HTML node

$(".asdf")[ 0 ] // the first HTML node
u(".asdf")[ 0 ] // error, should be u(".asdf").nodes[ 0 ]
````
These differences make it impossible for any plugin to support both jQuery and Umbrella.
Cash on the other hand seems like a valid alternative.

**Comment by phifa** [comment]
so you will drop jquery? i think that would be great. all major plugins seem to be moving away form jquery

**Comment by FrDH** [comment]
Yes. I'm currently working on version 8 (the [develop branche](https://github.com/FrDH/jQuery.mmenu/tree/develop)), it will be completely stand-alone.

**Comment by FrDH** [comment]
The [develop branche](https://github.com/FrDH/jQuery.mmenu/tree/develop) is about ready to be used. It no longer requires jQuery.
Please note that it still needs to be tested. It should work in Edge but I think you'll need polyfils for IE11.

**Comment by smth** [comment]
This is exciting. Any estimate on when the release will be?

**Comment by FrDH** [comment]
Somewhere in march I think.

**Comment by diemah77** [comment]
What is the right way to instantiate v8.0?

**Comment by FrDH** [comment]
````js
var menu = new Mmenu( "#my-menu", {
   // options
}, {
   // configs
});
````
For now, there is a jQuery wrapper (if jQuery or Zepto is defined), so this still works:
````js
$("#my-menu", {
   // options
}, {
   // configs
})
````

**Comment by diemah77** [comment]
Thanks!

**Comment by FrDH** [comment]
Update:

Version 8 has been released on NPM under the new plugin name:
https://www.npmjs.com/package/mmenu-js

I'm currently updating the website to match this new version. Once that's done, version 8 will become the master branche and version 7 (the jQuery plugin) will become the "legacy" brance.

**Comment by onlinebizsoft** [comment]
This is a great update. When do you think the version without jquery will be released to master on github?

**Comment by FrDH** [comment]
Before the end of this month.

**Comment by FrDH** [comment]
Here it is, version 8(.0.3) 

Please keep in mind that this new version targets modern browsers and IE11 will only work with polyfills.
For more information: http://www.mmenujs.com/documentation/
If you find any other polyfill to be needed, please let me know.
